update tiflash documents about cartesian join support (#5777)
pingcap/docs: tiflash/use-tiflash.md

* TopN: Performs the TopN calculation.
* Limit: Performs the limit calculation.
* Project: Performs the projection calculation.

> * HashJoin (having the join condition with the equivalent value): Performs the join calculation based on the [Hash Join](/explain-joins.md#hash-join) algorithm, but with the following conditions:

Reviewer: HashJoin(Equi Join)

Author: ```suggestion
* HashJoin (Equi Join): Performs the join calculation based on the [Hash Join](/explain-joins.md#hash-join) algorithm, but with the following conditions:
```

    * The operator can be pushed down only in the [MPP mode](#use-the-mpp-mode).
    * The push-down of `Full Outer Join` is not supported.

> * Hashjoin (Cartesian Join, which does not have the join condition with the equivalent value): Performs the Cartesian Join algorithm, but with the following conditions:

Reviewer 2: ```suggestion
* HashJoin (Cartesian Join, which does not have the join condition with the equivalent value): Performs the Cartesian Join algorithm, but with the following conditions:
```

> * HashJoin (Cartesian Join, which does not have the join condition with the equivalent value): Performs the Cartesian Join algorithm, but with the following conditions:

Reviewer: HashJoin(Non-Equi Join)

Author: ```suggestion
* HashJoin (Non-Equi Join): Performs the Cartesian Join algorithm, but with the following conditions:
```
